src/main/java/com/spotify/futures/ListenableToCompletableFutureWrapper.java

```diff
@@ -31,7 +32,7 @@
 
   ListenableToCompletableFutureWrapper(final ListenableFuture<V> future) {
     this.future = checkNotNull(future, "future");
-    Futures.addCallback(future, this);
+    Futures.addCallback(future, this, MoreExecutors.directExecutor());
```

Comment:
there are some reasons why Guava decided to deprecate all methods with implicit `directExecutor()`, may be it is a good chance to do the same here? Will it be a big changeset?

Reply:
Fixed for ConcurrencyLimiter, but kept for other use cases as the computation work done there is known/trivial so directExecutor is suitable.

Reply:
Can we move this logic out of the constructor while we are at it?

Reply:
I'll think about it, but I am not sure there exists any use case where you would want/need a different executor